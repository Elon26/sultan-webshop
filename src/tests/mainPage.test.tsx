import { render, screen } from "@testing-library/react";
import MainPage from "../components/pages/mainPage";
import { MemoryRouter } from "react-router-dom";

/** Тест проверяет наличие главной страницы в DOM-дереве, с сопуствующей проверкой основных её элементов на соответствие сохранённым snapshots'ам. */
describe("mainPage", () => {
    test("Проверка наличия в DOM-дереве и соответствия snapshots'ам", () => {
        render(
            <MemoryRouter>
                <MainPage />
            </MemoryRouter>
        );
        const header = screen.getByText(/Главная страница/i);
        const firstButton = screen.getByText(/Каталог/i);
        const secondButton = screen.getByText(/Админ-панель/i);
        expect(header).toBeInTheDocument();
        expect(firstButton).toBeInTheDocument();
        expect(secondButton).toBeInTheDocument();
        expect(header).toMatchSnapshot();
        expect(firstButton).toMatchSnapshot();
        expect(secondButton).toMatchSnapshot();
    });
});
