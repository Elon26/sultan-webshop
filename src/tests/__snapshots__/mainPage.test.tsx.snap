// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mainPage Проверка наличия в DOM-дереве и соответствия snapshots'ам 1`] = `
<h1
  class="pageTitle"
>
  Главная страница
</h1>
`;

exports[`mainPage Проверка наличия в DOM-дереве и соответствия snapshots'ам 2`] = `
<span>
  Каталог
</span>
`;

exports[`mainPage Проверка наличия в DOM-дереве и соответствия snapshots'ам 3`] = `
<span>
  Админ-панель
</span>
`;

exports[`mainPage Проверка наличия главной страницы в DOM-дереве, с сопуствующей проверкой основных элементов на соответствие сохранённым snapshots'ам 1`] = `
<h1
  class="pageTitle"
>
  Главная страница
</h1>
`;

exports[`mainPage Проверка наличия главной страницы в DOM-дереве, с сопуствующей проверкой основных элементов на соответствие сохранённым snapshots'ам 2`] = `
<span>
  Каталог
</span>
`;

exports[`mainPage Проверка наличия главной страницы в DOM-дереве, с сопуствующей проверкой основных элементов на соответствие сохранённым snapshots'ам 3`] = `
<span>
  Админ-панель
</span>
`;
